hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/JavaKeyStoreProvider.java

```diff
@@ -640,6 +640,10 @@ private void revertFromOld(Path oldPath, boolean fileExisted)
   private void renameOrFail(Path src, Path dest)
       throws IOException {
     if (!fs.rename(src, dest)) {
+      if (!fs.exists(src)) {
+        throw new FileNotFoundException(src.toUri().toString());
+      }
```

Comment:
How about keeping this check just before `fs#rename`? Could save efforts of doing rename in the first place?